### PR TITLE
chore: create target WireCountly similar to WireDatadog - WPB-11285

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11285" title="WPB-11285" target="_blank">TaskWPB-11285</a>  [iOS] Don't compile Countly SDK for C builds
  </td></table>
  <br />
 

### Issue

When trying to import WireAnalytics into WireDomain the project can currently not be built anymore (linker errors related to Countly).
This PR separates Countly from WireAnalytics.
A new target WireCountly is created similar to WireDatadog.
Via environment variable in the future Countly can be imported conditionally.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift [hunk 1]
```diff
@@ -1,94 +0,0 @@
...
-// You should have received a copy of the GNU General Public License
-// along with this program. If not, see http://www.gnu.org/licenses/.
-//
-
-import Foundation
-import WireDataModel
-
-// swiftlint:disable:next todo_requires_jira_link
-// TODO: move to DM
-private extension ZMConversation {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift:24]: why this file is removed?

File: wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift [hunk 2]
```diff
@@ -1,98 +0,0 @@
...
-
-import Foundation
-import WireDataModel
-
-enum ConversationType: Int {
-    case oneToOne
-    case group
-}
-
-extension ConversationType {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift:27]: same here ?

File: WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme [hunk 3]
```diff
@@ -1,112 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (netbe) [inline WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme:1]: why is this removed ? it's automatically generated by xcode right, is it ignored in gitignore too?

File: WireAnalytics/Sources/WarningPrevention/WarningPrevention.swift
Comment (netbe) [inline WireAnalytics/Sources/WarningPrevention/WarningPrevention.swift:1]: add a comment here too to state the purpose, if someone land in this file directly

// Prevents warning about not using dd....

File: WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift [hunk 4]
```diff
@@ -16,11 +16,11 @@
...
 //
 
-import Foundation
+public import Foundation
 
 // sourcery: AutoMockable
-/// Abstraction around `Countly`.
-protocol CountlyProtocol {
+/// Mirrors the API of `Countly`.
+public protocol CountlyAbstraction {
```
Comment (netbe) [inline WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift]: nit-pic: was this necessary to rename? at callsite we identify this as countlyProvider. Anyway, I don't have a strong opinion about it right now

File: wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift [hunk 5]
```diff
@@ -1,94 +0,0 @@
-//
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift:1]: Deleted unreferenced, unused file.

File: wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift [hunk 6]
```diff
@@ -1,98 +0,0 @@
-//
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift:1]: Deleted unreferenced, unused file.

File: .github/workflows/build_pr.yml [hunk 7]
```diff
@@ -24,6 +24,7 @@ jobs:
       ) && github.event.pull_request.draft == false
     uses: ./.github/workflows/_reusable_app_release_without_changelog.yml
     with:
+      countly_enabled: false
```
Comment (netbe) [inline .github/workflows/build_pr.yml:27]: 👍

File: .github/workflows/_reusable_app_release.yml [hunk 8]
```diff
@@ -7,6 +7,9 @@ on:
       is_cloud_build:
         type: boolean
         default: true
+      countly_enabled:
+        type: boolean
+        default: true
```
Comment (netbe) [inline .github/workflows/_reusable_app_release.yml]: you're missing c1_c2_c3_prod workflow. either we don't provide default here and we specify on every workflows that call it or we provide default but we skip the workflows that use the default

File: .github/workflows/c1_c2_c3_app_release_restricted.yml [hunk 9]
```diff
@@ -13,6 +13,10 @@ on:
           - all
         required: true
         default: 'all'
+      countly_enabled:
+        description: 'Enable Countly'
+        type: boolean
+        default: false
```
Comment (netbe) [inline .github/workflows/c1_c2_c3_app_release_restricted.yml]: don't expose the option and use false

**Review by netbe — CHANGES_REQUESTED** [review]
left a couple comments on workflows

File: .github/workflows/c1_c2_c3_app_release_restricted.yml (hunk 9 above)
Comment (caldrian) [reply .github/workflows/c1_c2_c3_app_release_restricted.yml]: fixed: 50b512451112562644363c5711c8bd14b3c0c2dd

File: .github/workflows/_reusable_app_release.yml (hunk 8 above)
Comment (caldrian) [reply .github/workflows/_reusable_app_release.yml]: fixed: ea9d155222c58c3f76230669051b3c17a3713aa9

File: WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme (hunk 3 above)
Comment (caldrian) [reply WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme:1]: I think it's only created when opening the package directly and not our workspace.
Shall we add them to the ignore file?

File: WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift (hunk 4 above)
Comment (caldrian) [reply WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift]: Wasn't necessary. But in my opinion the term abstraction fits better. It was even in the protocol doc comment.
Let's hear the opinion of the second reviewer.

File: wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/Analytics/Events/Analytics+Services.swift:24]: I left the comment on line 1.
It was an unreferenced file.

File: wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/Analytics/Events/ZMConversation+Analytics.swift:27]: Yes, same.
It was an unreferenced file.

File: WireAnalytics/Sources/WarningPrevention/WarningPrevention.swift
Comment (caldrian) [reply WireAnalytics/Sources/WarningPrevention/WarningPrevention.swift:1]: done: bad64cf605c3471e5936bdcfac7758ed3b2ac803

**Comment by samwyndham** [comment]
> Via environment variable in the future Countly can be imported conditionally.

Out of interest will this be a future requirement?

File: WireAnalytics/Package.swift [hunk 10]
```diff
@@ -4,53 +4,85 @@ import Foundation
...
+            sources: countlyFiles()
+        ),
+
+        .target(
+            name: "WireDatadog",
+            dependencies: datadogDependencies() + ["WireLogging"],
+            sources: datadogFiles()
+        ),
+
+        // This target prevents warnings about `countly-sdk-ios` or `dd-sdk-ios` not being used.
```
Comment (samwyndham) [inline WireAnalytics/Package.swift:52]: Sneaky!

File: WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift (hunk 4 above)
Comment (samwyndham) [reply WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift]: No strong opinion here. I would probably have kept with `Protocol`, just because that is what we generally use so far for similar objects. We could also use `Interface`. 🤯

**Review by samwyndham — APPROVED** [review]
Nice work!

File: WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift (hunk 4 above)
Comment (caldrian) [reply WireAnalytics/Sources/WireAnalytics/CountlyAbstraction/CountlyAbstraction.swift]: Reverted! 306c0596fb97fde2a49d3d619de70ea4c98314c6

File: WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme (hunk 3 above)
Comment (netbe) [reply WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme:1]: yes please, but I thought you did in some other PR. We can do in another one not to keep this one waiting

**Review by netbe — APPROVED** [review]
Good to go

File: WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme (hunk 3 above)
Comment (caldrian) [reply WireAnalytics/.swiftpm/xcode/xcshareddata/xcschemes/WireAnalytics-Package.xcscheme:1]: done here https://github.com/wireapp/wire-ios/pull/2232
